lib/modules/datasource/julia-pkg-server/registry.ts

```diff
@@ -0,0 +1,246 @@
+import { Parse as tarParse } from 'tar';
+import { logger } from '../../../logger';
+import type { Http } from '../../../util/http';
+import { Result } from '../../../util/result';
+import { joinUrlParts } from '../../../util/url';
+import {
+  PKG_SERVER_REQUEST_HEADERS,
+  juliaPkgServerDatasourceId,
+} from './common';
+
+/**
+ * Generic container for aggregating and passing around the contents of files.
+ */
+interface FileContents<T extends Buffer[] | string> {
+  [path: string]: T;
+}
+
+/**
+ * Wraps all the information necessary to uniquely identify a registry at a
+ * particular point in time.
+ */
+interface JuliaPkgServerRegistry {
+  pkgServer: string;
+  state: string;
+  uuid: string;
+}
+
+/**
+ * URLs referencing a registry served by a PkgServer take the form:
+ * `<PkgServer host>/registry/<registry UUID>/<registry state>`.
+ *
+ * Although required when querying the state of a registry, the state in the
+ * URL is not mandatory when configuring a `registryUrl` passed into the
+ * datasource. Therefore that section is marked optional and should be ensured
+ * to be filled before further processing.
+ */
+const PKG_SERVER_REGISTRY_URL_FORMAT = new RegExp(
```

Comment:
```suggestion
const PKG_SERVER_REGISTRY_URL_FORMAT = regEx(
```

Use regEx, which is re2 under the hood

Reply:
I had noticed that existed while starting to work on the versioning 😅